src/codewhisperer/activation.ts

```diff
@@ -259,49 +263,49 @@ export async function activate(context: ExtContext): Promise<void> {
                 await RecommendationHandler.instance.onCursorChange(e)
             }),
             vscode.workspace.onDidChangeTextDocument(async e => {
+                const editor = vscode.window.activeTextEditor
+                if (!editor) {
+                    return
+                }
+                if (e.document !== editor.document) {
+                    return
+                }
+                if (!runtimeLanguageContext.isLanguageSupported(e.document.languageId)) {
+                    return
+                }
+
                 /**
                  * CodeWhisperer security panel dynamic handling
                  */
-                if (e.document === vscode.window.activeTextEditor?.document) {
-                    disposeSecurityDiagnostic(e)
-                }
+                disposeSecurityDiagnostic(e)
 
                 CodeWhispererCodeCoverageTracker.getTracker(e.document.languageId)?.countTotalTokens(e)
 
                 /**
                  * Handle this keystroke event only when
-                 * 1. It is in current non plaintext active editor
-                 * 2. It is not a backspace
-                 * 3. It is not caused by CodeWhisperer editing
-                 * 4. It is not from undo/redo.
+                 * 1. It is not a backspace
+                 * 2. It is not caused by CodeWhisperer editing
+                 * 3. It is not from undo/redo.
                  */
-                if (
-                    e.document === vscode.window.activeTextEditor?.document &&
-                    runtimeLanguageContext.isLanguageSupported(e.document.languageId) &&
```

Comment:
just a sanity check if it's safe to remove this?

Reply:
The check is moved to above lines